### PR TITLE
added accept and reject routes for connection requests

### PR DESCRIPTION
**Pull request by matejAgilathon**

### CODE REVIEW

File: frontend/src/store/index.js [hunk 1]
```diff
@@ -117,6 +122,48 @@ export default new Vuex.Store({
         }
       })();
     },
+    acceptRequest(state) {
+      (async () => {
+        try {
+          const response = await axios.put(
+            `http://localhost:8000/api/connections/${state.connection.connectionId}`,
```
Comment (ivanodak1) [inline frontend/src/store/index.js]: Extract this endpoint to constant

File: frontend/src/store/index.js [hunk 2]
```diff
@@ -117,6 +122,48 @@ export default new Vuex.Store({
...
+            {
+              userId: state.user.userId,
+            },
+            {
+              withCredentials: true,
+            }
```
Comment (ivanodak1) [inline frontend/src/store/index.js:136]: This can fit in one line.

File: frontend/src/store/index.js [hunk 3]
```diff
@@ -117,6 +122,48 @@ export default new Vuex.Store({
...
+          console.log("Accepting connection request failed", error);
+          throw error;
+        }
+      })();
+    },
+    rejectRequest(state) {
+      (async () => {
+        try {
+          const response = await axios.delete(
+            `http://localhost:8000/api/connections/${state.connection.connectionId}`,
```
Comment (ivanodak1) [inline frontend/src/store/index.js]: same as above

File: frontend/src/store/index.js [hunk 4]
```diff
@@ -117,6 +122,48 @@ export default new Vuex.Store({
...
+            {
+              userId: state.user.userId,
+            },
+            {
+              withCredentials: true,
+            }
```
Comment (ivanodak1) [inline frontend/src/store/index.js:157]: same as above

File: frontend/src/views/UserView.vue [hunk 5]
```diff
@@ -9,12 +9,30 @@
     </div>
     <p class="profile-view__bio">{{ bio }}</p>
     <div class="profile-view__username">{{ username }}</div>
-    <button
-      class="profile-view__button"
-      @click="($event) => $store.commit('connect')"
+    <div v-if="$store.state.connection.connectionStatus === 'pending'">
```
Comment (ivanodak1) [inline frontend/src/views/UserView.vue]: extract this to computed/method. Also, do the null check before accessing connectionStatus

File: frontend/src/views/UserView.vue [hunk 6]
```diff
@@ -9,12 +9,30 @@
...
+      </button>
+      <button
+        class="profile-view__reject-button"
+        @click="($event) => $store.commit('rejectRequest')"
+      >
+        Reject
+      </button>
+    </div>
+    <div
+      v-else-if="$store.state.connection.connectionStatus === 'not-connected'"
```
Comment (ivanodak1) [inline frontend/src/views/UserView.vue]: same as above

**Review by ivanodak1 — CHANGES_REQUESTED** [review]

File: backend/controllers/connections/acceptConnection.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    const connectionUpdateRequest = await Connection.update(
+      {
+        status: "connected"
+      },
+      {
+        where: {
+          userId: Number(id),
+          connectionId: Number(userId),
+        },
+      }
```
Comment (ivanodak1) [inline backend/controllers/connections/acceptConnection.js]: Suggestion, extract this object argument
```
const options = {
        status: "connected"
      },
      {
        where: {
          userId: Number(id),
          connectionId: Number(userId),
        },
      }
```

File: frontend/src/store/index.js (hunk 2 above)
Comment (matejAgilathon) [reply frontend/src/store/index.js:136]: > This can fit in one line.

Prettier does it automatically

**Pull request merged**